pkg/action/install_test.go

```diff
@@ -361,7 +363,62 @@ func TestInstallRelease_Wait(t *testing.T) {
 	is.Contains(res.Info.Description, "I timed out")
 	is.Equal(res.Info.Status, release.StatusFailed)
 }
+func TestInstallRelease_Wait_Interrupted(t *testing.T) {
+	if os.Getenv("HANDLE_SIGINT") == "1" {
+		t.Run("Execute TestInstallRelease_Wait_Interrupted", func(t *testing.T) {
+			is := assert.New(t)
+			instAction := installAction(t)
+			instAction.ReleaseName = "interrupted-release"
+			failer := instAction.cfg.KubeClient.(*kubefake.FailingKubeClient)
+			failer.WaitDuration = 10 * time.Second
+			instAction.cfg.KubeClient = failer
+			instAction.Wait = true
+			vals := map[string]interface{}{}
+
+			res, err := instAction.Run(buildChart(), vals)
+			is.Error(err)
+			is.Contains(res.Info.Description, "SIGTERM or SIGINT received, release failed")
+			is.Equal(res.Info.Status, release.StatusFailed)
+		})
+		return
+
+	}
+	t.Run("Setup TestInstallRelease_Wait_Interrupted", func(t *testing.T) {
+		cmd := exec.Command(os.Args[0], "-test.run=TestInstallRelease_Wait_Interrupted")
```

Comment:
Why is the command re-executing itself? Why can't we run a function in the background using a `context` to test certain cancellation triggers?

https://pkg.go.dev/context

You can have `handleSignals()` return a `CancelFunc`, which can be provided to `Upgrade` as a `context.Context`. Then in test code you can either test `handleSignals` or provide an alternative context, like `context.WithDeadline()`.

Reply:
With the usage of context.Context this issue was removed